fleet/api/endpoint.go

```diff
@@ -47,6 +47,28 @@ func addAgentGroupEndpoint(svc fleet.Service) endpoint.Endpoint {
 	}
 }
 
+func viewAgentGroupEndpoint(svc fleet.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(viewResourceReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		agentGroup, err := svc.RetrieveAgentGroupByID(ctx, req.token, req.id)
+		if err != nil {
+			return nil, err
+		}
+		res := agentGroupRes{
+			ID:             agentGroup.ID,
+			Name:           agentGroup.Name.String(),
+			Description:    agentGroup.Description,
+			Tags:           agentGroup.Tags,
+			TsCreated:      agentGroup.Created,
+			MatchingAgents: map[string]interface{}{},
```

Comment:
This information is available in the view `agent_group_membership` but we will need another repo a method to retrieve it. Do you want to make that part of this PR or a future one?